src/pew/pygobject_gtk/__init__.py

```diff
@@ -87,16 +88,21 @@ def __init__(self):
 
         self.__gtk_application = Gtk.Application.new(
             self.application_id,
-            Gio.ApplicationFlags.FLAGS_NONE
+            Gio.ApplicationFlags.HANDLES_COMMAND_LINE | Gio.ApplicationFlags.HANDLES_OPEN
```

Comment:
Nice! The command line handling in particular I think would be really helpful for the other apps as well. 

My only question here would be if this is something that would be expected to be done universally by all apps using pyeverywhere, or would be app-specific. If the latter, I think it might be best to either have PEWApp flags for this (e.g. `handles_command_line=True`?) or to use the delegate pattern and check for the existence of  `handle_xyz` functions on the app's `PEWApp` subclass, and set these flags if found.

Reply:
Oh yeah, that could do with some explanation. GApplication includes its own command line handling, which for normal GTK apps means there are some common command line arguments like "--gtk-module" and "--gtk-name". Trouble with that is we'd need to use that library's own argument parsing stuff, which won't be very portable. So the [HANDLES_COMMAND_LINE](https://gjs-docs.gnome.org/gio20~2.0_api/gio.applicationflags#default-handles_command_line) flag turns it off. Instead _we_ are dealing with command lines. (It still does a helpful thing where if you try to run another instance of the application it passes that command line to the primary instance, but it doesn't do any extra processing).

This way an application is free to use argparse inside the _handle_command_line_ function and it should behave as usual. So PEW applications don't need to opt in or opt out. Simply leaving _handle_command_line_ unimplemented should keep the behaviour as usual. (This change really causes GTK applications to behave the same as others by default).

The _handle_open_file_uris_ function (and the [HANDLES_OPEN](https://gjs-docs.gnome.org/gio20~2.0_api/gio.applicationflags#default-handles_open) flag) is a bit weirder.

That is needed in kolibri-installer-gnome for the search provider: when an application is dbus-activated, the shell passes a selected URI that way instead of using the command line interface. (Indeed, I think there's an error in the documentation). A more file-centric PEW application might find a file URI in the _handle_command_line_ implementation, then pass that URI to _handle_open_file_uris_, which seems a bit silly at the moment. But there are some similar models for passing files to applications (at runtime) which could be supported with the same API.

Some poking around tells me setting this flag does affect how an app behaves, so I added a handles_open_file_uris flag which is off by default :)

Reply:
Okay, thanks for the detailed explanation! :) This all makes sense and looks good to me then!